src/components/GeoJSONLayer.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { useEffect, useState } from 'react'
+import { GeoJsonObject } from 'geojson'
+import { GeoJSONOptions } from 'leaflet'
+import { useMapInstance, GeoJSON } from '@datapunt/react-maps'
+
+type Props = {
+  url: string
+  options?: GeoJSONOptions
+}
+
+const GeoJSONLayer: React.FC<Props> = ({ url, options }) => {
+  const mapInstance = useMapInstance()
+  const [json, setJson] = useState<GeoJsonObject>()
+
+  useEffect(() => {
+    if (mapInstance) {
+      window
+        .fetch(`${url}`)
+        .then(res => res.json())
+        .then(res => {
+          setJson(res)
```

Comment:
Might not be a problem, but when the layer is removed this fetch will continue and eventually set the JSON after the component is destroyed.

We could cancel this fetch if the component is destroyed, see: https://developers.google.com/web/updates/2017/09/abortable-fetch